ui/ui-components/components/record/SampleRecordCard.tsx

```diff
@@ -0,0 +1,340 @@
+import Link from "next/link";
+import { Fragment, useCallback, useEffect, useMemo, useState } from "react";
+import { Button, Card, Col, Row, Table } from "react-bootstrap";
+import { useGrouparooModelContext } from "../../contexts/grouparooModel";
+import { ApiHook } from "../../hooks/useApi";
+import { usePrevious } from "../../hooks/usePrevious";
+import { Actions, Models } from "../../utils/apiData";
+import { successHandler } from "../../utils/eventHandlers";
+import { grouparooUiEdition } from "../../utils/uiEdition";
+import StateBadge from "../badges/StateBadge";
+import ManagedCard from "../lib/ManagedCard";
+import SeparatedItems from "../lib/SeparatedItems";
+import LinkButton from "../LinkButton";
+import AddSampleRecordModal from "./AddSampleRecordModal";
+
+interface Props {
+  execApi: ApiHook["execApi"];
+  properties: Models.PropertyType[];
+  disabled: boolean;
+}
+
+interface RecordRow {
+  propertyKey: string;
+  propertyId: string;
+  propertyState: Models.PropertyType["state"];
+  value: string | number | boolean;
+}
+
+const isConfigUI = grouparooUiEdition() === "config";
+
+const getCachedSampleRecordId = (modelId: string): string => {
+  return globalThis.localStorage?.getItem(`sampleRecord:${modelId}`);
+};
+
+const setCachedSampleRecordId = (modelId: string, recordId: string): void => {
+  recordId &&
+    globalThis.localStorage?.setItem(`sampleRecord:${modelId}`, recordId);
+};
+
+const clearCachedSampleRecordId = (modelId: string): void => {
+  globalThis.localStorage?.removeItem(`sampleRecord:${modelId}`);
+};
+
+const SampleRecordCard: React.FC<Props> = ({
+  properties,
+  execApi,
+  disabled,
+}) => {
+  const model = useGrouparooModelContext();
+  const prevModelId = usePrevious(model.id);
+
+  const [loading, setLoading] = useState(false);
+  const [importing, setImporting] = useState(false);
+  const [addingRecord, setAddingRecord] = useState(false);
+  const [hasRecords, setHasRecords] = useState(true);
+  const [record, setRecord] = useState<Actions.RecordView["record"]>();
+  const [groups, setGroups] = useState<Actions.RecordView["groups"]>();
+  const [recordId, setRecordId] = useState(() =>
+    getCachedSampleRecordId(model.id)
+  );
+
+  const saveRecord = useCallback(
+    (record, groups = undefined) => {
+      setLoading(true);
+      setRecord(record);
+      setGroups(groups);
+      setRecordId(record.id);
+      setHasRecords(true);
+      setLoading(false);
+
+      setCachedSampleRecordId(model.id, record.id);
+    },
+    [model]
+  );
+
+  const loadRecord = useCallback(async () => {
+    setLoading(true);
+
+    let record: Actions.RecordView["record"];
+    let groups: Actions.RecordView["groups"];
```

Comment:
The types for these is hard to resolve and I'd like to figure out a better way to do it. I need to revisit this to see if it's fixable.